### PR TITLE
simplify deploy using registry

### PR DESCRIPTION
**Pull request by jcantrill**
This PR:
* uses podman in place of imagebuilder
* simplies operator deployment with single deploy target
* fixes the location where to find catalog binaries

Note:  This works now but probably some cleanup required

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -87,9 +76,9 @@ fmt:
 lint: golangci-lint fmt
 	@golangci-lint run -c golangci.yaml
 
-image: imagebuilder
-	@if [ $${USE_IMAGE_STREAM:-false} = false ] && [ $${SKIP_BUILD:-false} = false ] ; \
-	then hack/build-image.sh $(IMAGE_TAG) $(IMAGE_BUILDER) $(IMAGE_BUILDER_OPTS) ; \
+image:
+	@if [ $${SKIP_BUILD:-false} = false ] ; then \
+		buildah bud -t $(IMAGE_TAG) . ; \
```
Comment (ewolinetz) [inline Makefile:81]: should we have a target for local dev where someone may not have `buildah` already on their system? or just let them resolve it separately (or add to docs?)

Comment (jcantrill) [reply Makefile:81]: IMO we should just document... you need the following tools to hack: x, y, z.

File: Makefile [hunk 2]
```diff
@@ -102,15 +91,18 @@ test-sec: gosec
 	@gosec -severity medium -confidence medium -exclude G304 -quiet ./...
 
 deploy: deploy-image
-	hack/deploy.sh
-.PHONY: deploy
+	LOCAL_IMAGE_ELASTICSEARCH_OPERATOR_REGISTRY=127.0.0.1:5000/openshift/elasticsearch-operator-registry \
+	$(MAKE) elasticsearch-catalog-build && \
+	IMAGE_ELASTICSEARCH_OPERATOR_REGISTRY=image-registry.openshift-image-registry.svc:5000/openshift/elasticsearch-operator-registry \
```
Comment (ewolinetz) [inline Makefile:96]: if we're setting this here, doesn't that make it so for local we aren't pushing to the `LOCAL_IMAGE_ELASTICSEARCH_OPERATOR_REGISTRY` ?

we set the port forwarding but then try to push to the remove registry...

Comment (jcantrill) [reply Makefile:96]: > if we're setting this here, doesn't that make it so for local we aren't pushing to the `LOCAL_IMAGE_ELASTICSEARCH_OPERATOR_REGISTRY` ?
> 
> we set the port forwarding but then try to push to the remove registry...

The original catalog build script does a build and push using the same env var.  The issue is, however, the generally you don't have access to the registry and thus the reason we portforward to localhost.  This provides a mechanism by which the CI should be satisfied along with our local dev experience

File: olm_deploy/operatorregistry/Dockerfile [hunk 3]
```diff
@@ -11,6 +11,6 @@ RUN chmod -R g+w /manifests && rm /manifests/0*.yaml /manifests/art.yaml /manife
 
 COPY olm_deploy/scripts/registry-init.sh /scripts/
 
-COPY --from=registry-builder /build/bin/initializer /usr/bin/initializer
-COPY --from=registry-builder /build/bin/registry-server /usr/bin/registry-server
+COPY --from=registry-builder /bin/initializer /usr/bin/initializer
```
Comment (ewolinetz) [inline olm_deploy/operatorregistry/Dockerfile:14]: nit -- you can probably rebase to drop this file's changes since we already fixed it

**Comment by bparees** [comment]
dev scripts only, overriding BZ req

**Pull request merged**